python/cudf/cudf/core/frame.py

```diff
@@ -114,6 +116,8 @@ class Frame:
     _index: Optional[cudf.core.index.BaseIndex]
     _names: Optional[List]
 
+    _VALID_BINARY_OPERATIONS = AllOperations
```

Comment:
Without knowing that `AllOperations` is a sentinel, it's easy to think that the `FOO`s and `BAR`s supported below are the same set:

```python
class Baz(Fooable, Barable):
    SUPPORTED_FOOS = AllOperations
    SUPPORTED_BARS = AllOperations
```

Can we consider some alternatives here?

* Instead of a sentinel value we use a concrete value bound to each type, like `_VALID_BINARY_OPERATIONS = BinaryOperand.ALL_OPERATIONS`. It's a bit more code, but it's more explicit.
* Default behaviour is to support all operations?

Reply:
Sure, that would be fine. We could store an instance of the `AllOperations` sentinel as a class member in each `OperationMixin` when it is created and the check would work just fine. 

I considered making the default supporting all operations, but I thought that might end up being more confusing. Consider something like
```
class A(Fooable):
    # Define the base operation, but not _SUPPORTED_FOOS

class B(A):
    _SUPPORTED_FOOS = {"a", "b"}
```
This is ill-defined since `B` should support everything that `A` does, but I think it's an easy mistake to make if you simply don't see `_SUPPORTED_FOOS` defined in `A`. By requiring an explicit opt-in you get something like 
```
class A(Fooable):
    # Define the base operation
    _SUPPORTED_FOOS=Fooable.AllOperations

class B(A):
    _SUPPORTED_FOOS = {"a", "b"}
```
and it's much easier to explain why specifying the operations for `B` is a no-op. What do you think?

Reply:
> Sure, that would be fine. We could store an instance of the AllOperations sentinel as a class member in each OperationMixin when it is created and the check would work just fine.

Instead of introducing a new sentinel into the mix, couldn't we just make `_ALL_FOOS` (a concrete set) an attribute of the mixin? We pass that into the `_create_delegating_mixin` factory anyway.

Reply:
Oh... yes. At that point, we could simply just use the existing `_SUPPORTED_FOOS` variable, right? We don't even need the alias.

Side note, we need to think of better names than "valid" and "supported". I have been frequently mixing them, and I wrote this code...

Reply:
Great call on that. Done.

Reply:
I think the comments here:
https://github.com/vyasr/cudf/blob/32c22d2d812fb97b5d927870285c01e1c53561d6/python/cudf/cudf/core/mixins/mixin_factory.py#L259-L261

Can be supplemented with the description related to this change. The attribute can be used to specify the full range of supported operators for the class.

Reply:
Good idea.